extensions/wikia/Wall/index/CommentsIndex.class.php

```diff
@@ -0,0 +1,255 @@
+	 * @return bool
+	 */
+	private function shouldUpdateParentInfoFor( CommentsIndexEntry $entry ): bool {
+		$oldEntry = $this->objectCache[$entry->getCommentId()];
+		if ( !( $oldEntry instanceof CommentsIndexEntry ) ) {
+			return false;
+		}
+
+		return
+			$entry->getParentCommentId() > 0 && (
+				$entry->isArchived() != $oldEntry->isArchived() ||
+				$entry->isDeleted() != $oldEntry->isDeleted()
+			)
+		;
+	}
+
+	/**
+	 * Insert a new Comments Index Entry into database
+	 * This assumes that wfReadOnly was called before and we're safe to write to master (see DAR-120 for details)
+	 * This has to be performed within the same transaction used by MediaWiki to save corresponding article comment to database,
+	 * to ensure data integrity.
+	 *
+	 * If this fails, transaction has to be rollbacked to prevent invalid data from being posted!
+	 *
+	 * @see https://wikia-inc.atlassian.net/browse/ZZZ-3225
+	 * @param CommentsIndexEntry $commentsIndexEntry
+	 * @param DatabaseBase $db
+	 * @return bool whether transaction succeeded
+	 */
+	public function insertEntry( CommentsIndexEntry $commentsIndexEntry, DatabaseBase $db ): bool {
+		$timestamp = $db->timestamp();
+
+		if ( empty( $commentsIndexEntry->getCreatedAt() ) ) {
+			$commentsIndexEntry->setCreatedAt( $timestamp );
+		}
+
+		if ( empty( $commentsIndexEntry->getLastTouched() ) ) {
+			$commentsIndexEntry->setLastTouched( $timestamp );
+		}
+
+		if ( empty( $commentsIndexEntry->getLastChildCommentId() ) ) {
+			$commentsIndexEntry->setLastChildCommentId( $commentsIndexEntry->getCommentId() );
+		}
+
+		// cache the new instance so we won't have to query the db when we need it again during this request
+		$this->objectCache[$commentsIndexEntry->getCommentId()] = $commentsIndexEntry;
+
+		$status = $db->replace(
+			'comments_index',
+			null,
+			$commentsIndexEntry->getDatabaseRepresentation(),
+			__METHOD__
+		);
+
+		if ( !$status ) {
+			$this->error( 'Failed to insert comments index entry', $commentsIndexEntry->getDatabaseRepresentation() );
+			return false;
+		}
+
+		// update last_child_comment_id for parent thread
+		if ( !empty( $commentsIndexEntry->getParentCommentId() ) ) {
+			$parentEntry = static::entryFromId( $commentsIndexEntry->getParentCommentId() );
+			$parentEntry->setLastChildCommentId( $commentsIndexEntry->getCommentId() );
+			$this->updateEntry( $parentEntry );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Return a single Comments Index Entry corresponding to the row in comments_index table with matching comment_id
+	 *
+	 * @param int $commentId
+	 * @return CommentsIndexEntry|null
+	 */
+	public function entryFromId( int $commentId ) {
+		if ( isset( $this->objectCache[$commentId] ) ) {
+			return $this->objectCache[$commentId];
+		}
+
+		$dbr = wfGetDB( DB_SLAVE );
+		$row = $dbr->selectRow( 'comments_index', '*', [ 'comment_id' => $commentId ], __METHOD__ );
+
+		// TODO: Monitor if this actually gets executed and remove if not
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this "TODO" comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)